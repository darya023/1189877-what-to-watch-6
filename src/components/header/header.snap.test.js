import React from 'react';
import {render, screen} from '@testing-library/react';
import {Router, Switch, Route} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import userEvent from '@testing-library/user-event';
import * as redux from 'react-redux';
import Header from './header.jsx';

const mockStore = configureStore({});

let history;

describe(`Test for Header`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`Header should render correctly`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: false,
        user: null,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <Header />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`Header should render correctly without User component`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: false,
        user: null,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <Header withoutUserComponent={true}/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`Header should render correctly with children`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: false,
        user: null,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <Header>
              <div>Test</div>
            </Header>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
  it(`Header should render correctly with children and without User component`, () => {
    const store = mockStore({
      USER: {
        authorizationStatus: false,
        user: null,
      },
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <Header withoutUserComponent={true}>
              <div>Test</div>
            </Header>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
