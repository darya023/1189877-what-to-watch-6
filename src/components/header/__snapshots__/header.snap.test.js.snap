// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header Header should render correctly 1`] = `
<div>
  <header
    class="page-header test"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
</div>
`;

exports[`Test for Header Header should render correctly with children 1`] = `
<div>
  <header
    class="page-header test"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div>
      Test
    </div>
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
</div>
`;

exports[`Test for Header Header should render correctly with children and without User component 1`] = `
<div>
  <header
    class="page-header test"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div>
      Test
    </div>
    
  </header>
</div>
`;

exports[`Test for Header Header should render correctly without User component 1`] = `
<div>
  <header
    class="page-header test"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    
  </header>
</div>
`;
